javascript/diff_match_patch_uncompressed.js

```diff
@@ -2412,16 +2412,22 @@ diff_match_patch.patch_obj.prototype.toString = function() {
 
 // CLOSURE:begin_strip
 // Lines below here will not be included in the Closure-compatible library.
-
-// Export these global variables so that they survive Google's JS compiler.
-// In a browser, 'this' will be 'window'.
-// Users of node.js should 'require' the uncompressed version since Google's
-// JS compiler may break the following exports for non-browser environments.
-/** @suppress {globalThis} */
-this['diff_match_patch'] = diff_match_patch;
-/** @suppress {globalThis} */
-this['DIFF_DELETE'] = DIFF_DELETE;
-/** @suppress {globalThis} */
-this['DIFF_INSERT'] = DIFF_INSERT;
-/** @suppress {globalThis} */
-this['DIFF_EQUAL'] = DIFF_EQUAL;
+if (typeof module === 'object') {
+  // The following export code was added by @ForbesLindesay
+  module.exports = diff_match_patch;
+  module.exports['diff_match_patch'] = diff_match_patch;
+  module.exports['DIFF_DELETE'] = DIFF_DELETE;
+  module.exports['DIFF_INSERT'] = DIFF_INSERT;
+  module.exports['DIFF_EQUAL'] = DIFF_EQUAL;
+} else {
```

Comment:
@TheSpyder you've been so persistent and so helpful on this. I wish I had more ability to quickly respond.

in reflection I think it's best for us to lean hard into ES Modules. let's eliminate all of the CommonJS references and only create ESM output.

I've been wanting to do this myself instead of asking you to make the change, but that has only resulted in me deferring this work. so, if you get this updated we'll get it merged, otherwise you can still wait on me and I'll merge it when I can.

Reply:
@TheSpyder I've pushed a commit and may have messed up everything. Would you be interested in confirming if my changes work for your project?

I haven't fixed the tests, and I don't like that, but I'm willing to let that remain a follow-up issue because I want to overhaul the tests anyway; take them out of the browser and into a more raw test runner.

Reply:
That's fine! I understand. I'll take a look soon, Monday at the latest.

I have already converted the JS tests to a nodejs runner, that's what I pulled out of this PR at your request. Once it's cleaned up and merged that will go into PR as well 🙂

Reply:
yeah sorry. I'm horribly inconsistent and slow at times.

if you want, throw out what I just pushed - which is the start of rewriting the tests. I'd almost prefer we do all the tests in a different change, but I did want to make sure it's viable to make the switch, so tossed out Bun and a standard runner/spec file.

let me know your thoughts.

Reply:
I used zora which is extremely lightweight. I know Jest is popular, but I find it too heavy for my liking :)

My commit is here:
https://github.com/TheSpyder/diff-match-patch/commit/3d8b914b05d966e2f31a5810b468ae7e564da237

It focuses on minimal changes in the diff, to preserve the tests themselves and only need reviewing on the test runner.

Reply:
> yeah sorry. I'm horribly inconsistent and slow at times.

I will never judge you for taking your time, or changing your mind. I fully understand the pressures of maintaining something open source like this 🙂

Reply:
For tests, [have you considered the node test runner?](https://nodejs.org/docs/latest-v20.x/api/test.html#test-runner)

Reply:
Thanks @sirreal, I forgot that `node` itself has a runner. I think that would be best and I chose `bun` specifically because it didn't require installing a separate testing library.

Reply:
I'll take a look. A library can make writing tests easier, but I don't know that we will be adding new tests very often.

Reply:
I'm going to revert your changes @dmsnell and redo them - I'd like to make sure I can prove the diff made as few edits as possible, which means avoiding code formatting. We can apply that later. This will also force push away the bun/jest code while I investigate swapping to the node test runner.

Reply:
I saved a local copy of the branch in case we need it.

Reply:
So of course after all that I have changed to webstorm since opening this PR and it reformatted some code 🤦 sorry about the extra force pushes.

Reply:
I know, but not everyone does 😂